src/SeaIceDynamics/Rheologies/explicit_visco_plastic_rheology.jl

```diff
@@ -0,0 +1,278 @@
+- `yield_curve_eccentricity`: eccentricity of the elliptic yield curve. Default `2`.
+- `Δ_min`: Minimum value for the visco-plastic parameter. Limits the maximum viscosity of the ice, 
+           transitioning the ice from a plastic to a viscous behaviour. Default value is `1e-10`.
+- `min_substeps`: Minimum number of substeps expressed as the dynamic coefficient. Default value is `30`.
+- `max_substeps`: Maximum number of substeps expressed as the dynamic coefficient. Default value is `500`.
+"""
+function ExplicitViscoPlasticRheology(FT::DataType = Float64; 
+                                      ice_compressive_strength = 27500, 
+                                      ice_compaction_hardening = 20, 
+                                      yield_curve_eccentricity = 2, 
+                                      Δ_min = 2e-9,
+                                      min_substeps = 30,
+                                      max_substeps = 500)
+
+    return ExplicitViscoPlasticRheology(convert(FT, ice_compressive_strength), 
+                                        convert(FT, ice_compaction_hardening), 
+                                        convert(FT, yield_curve_eccentricity),
+                                        convert(FT, Δ_min),
+                                        convert(FT, min_substeps),
+                                        convert(FT, max_substeps))
+end
+
+function required_auxiliary_fields(grid, ::ExplicitViscoPlasticRheology)
+    
+    # TODO: What about boundary conditions?
+    σ₁₁ = Field{Center, Center, Nothing}(grid)
+    σ₂₂ = Field{Center, Center, Nothing}(grid)
+    σ₁₂ = Field{Face, Face, Nothing}(grid)
+    
+    uⁿ = Field{Face, Center, Nothing}(grid)
+    vⁿ = Field{Center, Face, Nothing}(grid)
+    P  = Field{Center, Center, Nothing}(grid)
+
+    substeps = Field{Face, Face, Nothing}(grid) # Dynamic substeps a la Kimmritz et al (2016)
+
+    # An initial (safe) educated guess
+    fill!(substeps, 100)
+
+    return (; σ₁₁, σ₂₂, σ₁₂, substeps, uⁿ, vⁿ, P)
+end
+
+# Extend the `adapt_structure` function for the ExplicitViscoPlasticRheology
+Adapt.adapt_structure(to, r::ExplicitViscoPlasticRheology) = 
+    ExplicitViscoPlasticRheology(Adapt.adapt(to, r.ice_compressive_strength),
+                                 Adapt.adapt(to, r.ice_compaction_hardening),
+                                 Adapt.adapt(to, r.yield_curve_eccentricity),
+                                 Adapt.adapt(to, r.Δ_min),
+                                 Adapt.adapt(to, r.min_substeps),
+                                 Adapt.adapt(to, r.max_substeps))
+
+@inline function fill_rheology_halo_regions!(solver, ::ExplicitViscoPlasticRheology) 
+    fill_halo_regions!(solver.auxiliary_fields.σ₁₁)
+    fill_halo_regions!(solver.auxiliary_fields.σ₁₂)
+    fill_halo_regions!(solver.auxiliary_fields.σ₂₂)
+    fill_halo_regions!(solver.auxiliary_fields.substeps)
```

Comment:
so weird

Reply:
I can remove this and fill halo in `compute_stresses!`